MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -51,6 +51,18 @@ EFI_MEMORY_ATTRIBUTE_PROTOCOL  *MemoryAttributeProtocol = NULL;
 
 #define SPECIAL_REGION_PATTERN  7732426 // 7-(S) 7-(P) 3-(E) 2-(C) 4-(I) 2-(A) 6-(L)
 
+// TRUE if A and B have overlapping intervals
+#define CHECK_OVERLAP(AStart, AEnd, BStart, BEnd) \
+  ((AStart <= BStart && AEnd > BStart) || \
+  (BStart <= AStart && BEnd > AStart))
+
+// TRUE if A interval subsumes B interval
+#define CHECK_SUBSUMPTION(AStart, AEnd, BStart, BEnd) \
+  ((AStart < BStart) && (AEnd > BEnd))

```

Comment:
double check my logic here, but doens't Aend and Bend need to be >= instead of just >?

Reply:
Technically true, but if AEnd == BEnd, we don't need to create a new special region entry to cover BEnd->AEnd which is how this macro is used.